pkg/storer/internal/cache/cache.go

```diff
@@ -232,7 +232,9 @@ func (c *Cache) RemoveOldest(ctx context.Context, st transaction.Storage, count
 				if err != nil {
 					return err
 				}
-				c.size.Add(-1)
+				if c.size.Load() > 0 {
```

Comment:
instead of doing this here: I would make the check here https://github.com/ethersphere/bee/blob/master/pkg/storer/internal/cache/cache.go#L64

but we need to understand why the size went below zero

Reply:
or maybe it did not go down below zero, and the cache size was too large to begin with?

Reply:
Looking at the panic, 
```
github.com/ethersphere/bee/v2/pkg/storer/internal/cache.(*Cache).RemoveOldest(0x140025803d8, {0x101fa2fc0, 0x140001cc780}, {0x101fa6060, 0x1400253c2a0}, 0xffffffffffffffff)
```

`RemoveOldest` was called with a `count = 0xffffffffffffffff`. That's max uint64. 
This indicates that the size went to -1.


When `RemoveOldest ` is called, `len(evictItems)` is not always equal to the cache size at that instant because of the `put` happening in another goroutine.

```
fmt.Println("len of evict items", len(evictItems), "cache size", c.size.Load())

> len of items to evict 10, cache size 10, size after 0
> len of items to evict 1, cache size 6, size after 5
> len of items to evict 6, cache size 5, size after -1 
```